ui/pages/create-account/connect-hardware/select-hardware.js

```diff
@@ -51,10 +51,28 @@ export default class SelectHardware extends Component {
 
   state = {
     selectedDevice: null,
+    trezorRequestDevicePending: false,
   };
 
-  connect = () => {
+  connect = async () => {
     if (this.state.selectedDevice) {
+      if (this.state.selectedDevice === 'trezor') {
+        this.setState({ trezorRequestDevicePending: true });
+        try {
+          await window.navigator.usb.requestDevice({
+            filters: [
+              { vendorId: 0x534c, productId: 0x0001 },
+              { vendorId: 0x1209, productId: 0x53c0 },
+              { vendorId: 0x1209, productId: 0x53c1 },
+            ],
+          });
+          this.setState({ trezorRequestDevicePending: false });
+        } catch (e) {
+          this.setState({ trezorRequestDevicePending: false });
+          throw e;
+        }
```

Comment:
nite: something like the following is slightly cleaner

```suggestion
        } catch (e) {
          throw e;
        } finally {
          this.setState({ trezorRequestDevicePending: false });
        }
```

Reply:
Addressed !